test/bdm_interface_test.h

```diff
@@ -0,0 +1,28 @@
+#include "bdm_interface.h"
```

Comment:
Please rename this file to `test/bdm_interface_test.cc`
Since it is not a source file and not included in another translation unit, these tests are not compiled nor executed at the moment.
Renaming will fix this issue